file/writable_file_writer.cc

```diff
@@ -451,6 +564,73 @@ IOStatus WritableFileWriter::WriteBuffered(const char* data, size_t size) {
     src += allowed;
   }
   buf_.Size(0);
+  buffered_data_crc32c_checksum_ = 0;
+  return s;
+}
+
+IOStatus WritableFileWriter::WriteBufferedWithChecksum(const char* data,
+                                                       size_t size) {
+  IOStatus s;
+  assert(!use_direct_io());
+  assert(perform_data_verification_ && buffered_data_with_checksum_);
+  const char* src = data;
+  size_t left = size;
+  DataVerificationInfo v_info;
+  char checksum_buf[sizeof(uint32_t)];
+
+  // Check how much is allowed
+  size_t data_size = left;
+  if (rate_limiter_ != nullptr) {
+    while (data_size > 0) {
```

Comment:
I guess we loop here until the rate limiter allows to write the entire buffer since the checksum is calculated on the whole buffer. Maybe we can revisit at some later point, since it sort of defeats the purpose of the rate limiter.

Reply:
Sure. I will add a TODO to indicate the potential improvement.